dataset.py

```diff
@@ -1,56 +1,18 @@
-# 
-'''
-NOTE: Only testing on Mountain Car for now, so refactoring/modifying car_dset.py
-
-Data structure of the input .npz:
-the data is save in python dictionary format with keys: 'acs', 'ep_rets', 'rews', 'obs'
-the values of each item is a list storing the expert trajectory sequentially
-a transition can be: (data['obs'][t], data['acs'][t], data['obs'][t+1]) and get reward data['rews'][t]
-'''
-
 import numpy as np
-import argparse
-import os
-
-class Data(object):
-    """helper class to manage batches"""
-
-    def __init__(self, inputs, labels, randomize):
-        self.inputs = inputs
-        self.labels = labels
-        assert len(self.inputs) == len(self.labels)
-        self.randomize = randomize
-        self.num_pairs = len(inputs)
-        self.init_pointer()
-
-    def init_pointer(self):
-        self.pointer = 0
-        if self.randomize:
-            idx = np.arange(self.num_pairs)
-            np.random.shuffle(idx)
-            self.inputs = self.inputs[idx, :]
-            self.labels = self.labels[idx, :]
-
-    def get_next_batch(self, batch_size):
-        # if batch_size is negative -> return all
-        if batch_size < 0:
-            return self.inputs, self.labels
-        if self.pointer + batch_size >= self.num_pairs:
-            self.init_pointer()
-        end = self.pointer + batch_size
-        inputs = self.inputs[self.pointer:end, :]
-        labels = self.labels[self.pointer:end, :]
-        self.pointer = end
-        return inputs, labels
 
-def split_by_quantile(data, q, env_name='Hopper-v2'):
+def split_by_quantile(data, q, env_name='Hopper-v2', proxy='mean_padding'):
     """splits the data according to the quantile q of the Dataset"""
     
     if env_name == 'MountainCar-v0':
         sum_positions = data['obs'][:,:,0].sum(axis=-1)
         furthest_right = np.argsort(sum_positions)
     elif env_name == 'Hopper-v2':
-        furthest_right = np.argsort([np.sum(forward_traj) for forward_traj in      data['obs'][:,:,4]])
+        if proxy == 'mean':
```

Comment:
maybe leave a comment to say "this is the same as proxy, because length is constant"

Reply:
Happy to leave more comments!

I think there is a confusion here though. I used to do `np.sum` (cf. deleted line), but now I'm doing the 'mean_padding` thing on line 12-14 that doesn't consider that the length is constant, but still does a mean on the non-zero values.

But yeah, on slack I said that doing only `np.sum` (what I was doing before) is equivalent to the np.mean [here](https://github.com/RyanCarey/quantilizers-muj/blob/master/split_data.py#L17) you did for your experiments (because the length is constant).